nca/CoreDS/ConnectionSet.py

```diff
@@ -105,10 +105,13 @@ def _aggregate_connection_representation(protocols):
         protocols_not_aggregated = protocols
         aggregation_results = ''
 
-        # handle TCP+UDP properties aggregation (do not handle range segmentation overlapping)
+        # handle TCP+UDP ports aggregation (do not handle range segmentation overlapping)
         tcp_protocol_number = ProtocolNameResolver.get_protocol_number('TCP')
         udp_protocol_number = ProtocolNameResolver.get_protocol_number('UDP')
-        if protocols_not_aggregated.get(tcp_protocol_number) and protocols_not_aggregated.get(udp_protocol_number):
+        tcp_protocol = protocols_not_aggregated.get(tcp_protocol_number)
+        udp_protocol = protocols_not_aggregated.get(udp_protocol_number)
+        if tcp_protocol and udp_protocol and tcp_protocol.active_dimensions and udp_protocol.active_dimensions and \
+                all(i in ['dst_ports', 'src_ports'] for i in tcp_protocol.active_dimensions):
```

Comment:
```suggestion
        if tcp_protocol and udp_protocol and tcp_protocol.active_dimensions and udp_protocol.active_dimensions == tcp_protocol.active_dimensions:
```
Can simplify this expression.
`active_dimensions` has deterministic order.

Reply:
Done

Reply:
You can also remove `and udp_protocol.active_dimensions` from this expression , since you already check that for `tcp_protocol.active_dimensions` .